examples/rocshmem_allreduce_test.cc

```diff
@@ -123,16 +123,14 @@ int main (int argc, char **argv)
         nelem = atoi(argv[1]);
     }
 
-    int my_pe = rocshmem_my_pe();
-    int npes =  rocshmem_n_pes();
-
-    int ndevices, my_device = 0;
-    CHECK_HIP(hipGetDeviceCount(&ndevices));
-    my_device = my_pe % ndevices;
-    CHECK_HIP(hipSetDevice(my_device));
+    char* ompi_local_rank = getenv("OMPI_COMM_WORLD_LOCAL_RANK");
```

Comment:
I would prefer if we would introduce a function in a new header file in the examples directory (which is included by all files in this directory) that returns the local rank. The initial version of that routine can do the `getenv("OMPI_COMM_WORLD_LOCAL_RANK")`, but we might add additional mechanisms later, e.g. for MPICH/CrayMPICH and SLURM (they have separate environment variables)